instrumentation/opentelemetry-instrumentation-grpc/src/opentelemetry/instrumentation/grpc/_client.py

```diff
@@ -52,17 +51,6 @@ def release(self):
         return self.span
 
 
-def _inject_span_context(metadata: MutableMapping[str, str]) -> None:
-    # pylint:disable=unused-argument
-    def append_metadata(
-        carrier: MutableMapping[str, str], key: str, value: str
-    ):
-        metadata[key] = value
```

Comment:
sorry just to clarify: what is now being done differently in the code?

I see that `opentelemetry.propagators.inject` uses the `default_setter` by default, which looks to just also set the value. 

There is a weird thing in this method where carrier is not used, but the value should be equal (`metadata`)

Reply:
The API was updated to use a default setter so that carriers that are actually dictionaries (majority of cases) don't have to supply their own setters. This instrumentation was not updated to use the new API probably because it did not have a test for propagation. This PR adds a test and updates the propagation API usage.

Reply:
I opened this issue recently https://github.com/open-telemetry/opentelemetry-python-contrib/issues/382

I think we need to keep a custom setter and actually lowercase the keys here. gRPC throws an exception if metadata has any non-lowercase characters

Reply:
I see. I'll add it back and fix the inject call signature. Thanks @aabmass